batch/batch/front_end/front_end.py

```diff
@@ -1054,9 +1046,10 @@ async def ui_get_job(request, userdata):
 WHERE batch_id = %s AND job_id = %s
 ''',
             (batch_id, job_id))]
```

Comment:
Hmm. I feel like we should get all the information in one query:

```sql
SELECT jobs.batch_id, jobs.job_id, attempts.*
FROM jobs
INNER JOIN batches ON jobs.batch_id = batches.id
LEFT JOIN attempts ON jobs.batch_id = attempts.batch_id and jobs.job_id = attempts.job_id
 WHERE user = %s AND jobs.batch_id = %s AND NOT deleted AND jobs.job_id = %s;
```
If a job has no attempts, you'll get one record with all fields set to `None`. If there are one or more attempts, you'll get one or more records.

So then I think the not found check is:
```python3
attempts = [x async for x in db.select_and_fetchall(...)]
if len(attempts) == 0:
    raise web.HTTPNotFound()
if len(attempts) == 1 and attempts.attempt_id is None:
    raise web.HTTPNotFound()
for attempt in attempts:
    ...
```

Reply:
I thought it wouldn't do what I wanted with a left join and would only keep one attempt, but I verified with a Fiddle that it does the correct thing.